_posts/2021-12-20-solidity-0.8.11-release-announcement.md

```diff
@@ -0,0 +1,89 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.8.11 Release Announcement'
+date: '2021-12-20'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.8.11](https://github.com/ethereum/solidity/releases/tag/v0.8.11) ...
+
+## Notable New Features
+
+### Language Server Preview
+
+[Language Server Protocol](https://microsoft.github.io/language-server-protocol/)
+is an initiative that allows better interoperability between IDEs and
+compilers, or more general, language diagnostics tools. The idea is that
+instead of writing a different plugin for each combination of IDEs and
+languages, every IDE implements a single client and every compiler implements
+a single server and since both of them use the same protocol, you can combine
+all of them with each other.
+
+Solidity 0.8.11 contains a first release of an integrated language server,
+which means every IDE that has Language Server support will support Solidity
+out of the box.
+
+Since we first want to focus on getting the interface right, this release
+contain only a minimal set of features: It can only report errors and warnings.
+
+Having said that, we expect future features to be finished quickly. In the first
+quarter of 2022 we will probably have "jump to definition", "rename", "find all references",
+"show documentation on hover", "semantic hilighting" and more.
+
+Unfortunately, the one feature that will take more effort is the one you were
+probably looking forward to the most: autocomplete. The reason is that
+autocomplete requires the compiler to perform symbol lookup and type resolution
+inside broken (because it is incomplete) code. We have a prototype that can handle
+incomplete code, but we do not yet know how useful it is.
+
+#### How to Use the LSP
+
+How to hook your IDE up with the Solidity LSP of course depends on your IDE.
+The LSP is currently only implemented for the native binary (i.e. it does not
+work if you install Solidity via javascript / npm). You need to download
+the binary and provide the path to your IDE, plus the option ``--lsp``.
+
+**QtCreator:** Choose Tools -> Options -> Language Client. Click on "Add" and
+provide the path to the solc executable and ``--lsp`` as options.
+
+**vim / neovim:** You need the ``coc`` plugin and add the following to the
+``coc-settings.json`` file:
+
+```
+{
+    // ...
+    "languageserver": {
+        // ...
+        "solidity": {
+            "command": "/path/to/solc",
+            "args": [ "--lsp" ],
+            "rootPatterns": [".git/"],
+            "filetypes": ["solidity"]
+        }
+    }
+}
+```
+
+**VS Code:** TODO
+
+#### Please Provide Feedback!
+
+Which of the future features would you find most important?
+Does the LSP work for you? Is it easy to install?
+
+TODO where to provide feedback
```

Comment:
We could create a new type of issue template that people can click which is related to solc LSP feedback and upon ticket creation a `lsp-feedback` would be applied automatically.